tests/Unit/NumericalAlgorithms/Interpolation/Python/Test_RegularGridInterpolant.py

```diff
@@ -0,0 +1,89 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import unittest
+from spectre import Spectral
+from spectre import DataStructures
+from spectre import Interpolation
+from numpy.polynomial.legendre import Legendre
+import numpy as np
+
+
+class TestRegularGrid(unittest.TestCase):
+
+    # random polynomial functions of low order for exact interpolation
+    def polynomial(self, coords):
+        dim = len(coords)
+        if dim == 1:
+            x = coords[0]
+            return x**2 + x + 1
+
+        elif dim == 2:
+            x, y = coords
+            return 2 * x**2 + y**2 + y + x + 2
+
+        elif dim == 3:
+            x, y, z = coords
+            return 3 * x**2 + 2 * y**2 + z**2 + 2 * y + z + x + 2
```

Comment:
[stylistic:] I think we prefer to write floats as "3.0" or "3." but not "3"... this applies to the prefactors on the various terms (but not the exponents which are integers). Same for 1D and 2D lines above.